feat(netbox): Add NetBox Community

TheDodger opened the pull request:
**Description**
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->
Please incubate.


⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [X] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code
- [ ] 📜 Documentation Changes

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->
Running nicely.

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [X] ⚖️ My code follows the style guidelines of this project
- [X] 👀 I have performed a self-review of my own code
- [X] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made changes to the documentation
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning
- [X] I made sure the title starts with `feat(chart-name):`, `fix(chart-name):`, `chore(chart-name):`, `docs(chart-name):` or `fix(docs):`

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png` (not yet)


---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

TheDodger committed:
Add Netbox

TheDodger committed:
Remove CNPG backup config

TheDodger committed:
Nicer housekeeping & OSM

TheDodger committed:
Simplify

TheDodger committed:
Fix DB connection

TheDodger committed:
Fix Redis creds

TheDodger committed:
Revert a bit

TheDodger committed:
Shuffle around for better oversight of configurable options

TheDodger committed:
Add CI stuff

alfi0812 commented on charts/incubator/netbox/values.yaml:
```diff
@@ -0,0 +1,349 @@
+            SECRET_KEY:
+              secretKeyRef:
+                name: netbox-secrets
+                key: SECRET_KEY
```
Please also remove the auto generated secret and let people just enter one themselves as a value. Otherwise backup and restore and database backups might not work correctly if people dont use the same secrets

TheDodger replied:
The secret just gets used at runtime, there's no need to set a fixed one and Netbox refuses to start if too short. So better keep it that way.

https://netbox.readthedocs.io/en/stable/configuration/required-parameters/#secret_key

alfi0812 replied:
> The secret just gets used at runtime, there's no need to set a fixed one and Netbox refuses to start if too short. So better keep it that way.
> 
> https://netbox.readthedocs.io/en/stable/configuration/required-parameters/#secret_key

No. Remove it. We have multiple examples of just using with values.
We dont want tpl where its not needed and it goes against beeing replicable. Which a random secret isnt.

alfi0812 replied:
> NetBox deployments comprising multiple nodes must have the same secret key configured on all nodes.

Random secrets would also break this option. (If not in the same cluster)

alfi0812 replied:
Can add a comment about its requirements and/or how to generate it

Crow-Control commented on charts/incubator/netbox/values.yaml:
```diff
@@ -0,0 +1,349 @@
+        ### plugins.py content goes here
+
+        PLUGINS = [
+            'plugin1',
+            'plugin2',
```
isn't this going to cause issues?

TheDodger replied:
Nope, its mapping is disabled by default, see line 164 and 171.

Crow-Control commented on charts/incubator/netbox/templates/_secrets.tpl:
```diff
@@ -0,0 +1,13 @@
+enabled: true
+data:
+  SECRET_KEY: {{ $secretKey }}
+{{- end -}}
```
We dont want tpls files unless its absolutely needed

alfi0812 replied:
Already told them above

Crow-Control commented on charts/incubator/netbox/templates/_secrets.tpl:
```diff
@@ -0,0 +1,13 @@
+
+ {{- with lookup "v1" "Secret" .Release.Namespace $secretName -}}
+   {{- $secretKey = index .data "SECRET_KEY" | b64dec -}}
+ {{- end }}
```
lookups are forbidden, due to compatibility issues

alfi0812 replied:
Its not needed at all. So if they remove them this is also fixed

Crow-Control commented on charts/incubator/netbox/templates/_secrets.tpl:
```diff
@@ -0,0 +1,13 @@
+{{- define "netbox.secrets" -}}
+{{- $secretName := (printf "%s-netbox-secrets" (include "tc.v1.common.lib.chart.names.fullname" $)) }}
+
+{{- $secretKey := randAlphaNum 64 -}}
```
No lookups, means random is also forbidden.

alfi0812 replied:
Its not needed at all. So if they remove them this is also fixed

Crow-Control commented on charts/incubator/netbox/CHANGELOG.md:
```diff
@@ -0,0 +1,4 @@
+---
+title: Changelog
+pagefind: false
+---
```
Dont, this is autogenerated

Crow-Control commented on charts/incubator/netbox/ci/basic-values.yaml:
```diff
@@ -0,0 +1,6 @@
+    probeTimeouts:
+      startup:
+        initialDelaySeconds: 30
+        failureThreshold: 120
```
If this is needed for tests, its needed outside of tests as well.

alfi0812 replied:
I suggest trying to add more resources first before trying this.

TheDodger replied:
I'm not done playing with this and limits. Vanilla image starts flawlessly on my cluster.

Not so with my custom one and (nearly) all plugins enabled, this still gets killed after some time.

TheDodger committed:
Cleanup

Get rid of secret keygen

Raise limits

Add initContainer for worker

TheDodger committed:
Add initialDelaySeconds to values.yaml

Cleanup

TheDodger committed:
Shave off a few secs

spamming the logs, to (hopefully) get this through as the last run nearly finished

TheDodger committed:
Lets give this a try

Crow-Control commented on charts/incubator/netbox/ci/test-values.yaml:
```diff
@@ -0,0 +1,7 @@
+  limits:
+    cpu: 4000m
+  requests:
+    cpu: 1000m
```
This needs to be our default ratio as well btw.

alfi0812 commented on charts/incubator/netbox/values.yaml:
```diff
@@ -0,0 +1,365 @@
+    type: emptyDir
+    medium: Memory
+    mountPath: "/opt/unit/state"
+securityContext:
```
Why cant this run with our default 568?

TheDodger replied:
TrueNAS legacy apps user? The Netbox container uses 1000 by default.

alfi0812 replied:
Has nothing to do with truenas. Its the default apps user for Linux and also what we use for all of our charts

alfi0812 replied:
Its fine if required but shouldn't be. And 568 should work just fine

alfi0812 commented on charts/incubator/netbox/values.yaml:
```diff
@@ -0,0 +1,365 @@
+            limits:
+              cpu: 2000m
+            requests:
+              cpu: 200m
```
And this will overwrite your global ci value. And only allow 0.2 threads for this container

TheDodger replied:
Are you kidding me? That's the minimum requested, `limits` defines the max. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits

But I'll look into it.

alfi0812 replied:
> Are you kidding me? That's the minimum requested, `limits` defines the max. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits
> 
> But I'll look into it.

2 max and 0,2 requests. But those make your ci values useless as the per pod settings stil wont allow more

alfi0812 replied:
Also Just because we decline something or request changes is no reason to start getting rude. 

If you limit a pod to certain restrains, global options will only set them for every other pod. So clearly a mistake. 
Thats what i meant. I do know how requests and limits do work. But i also know how our common works with global and per pod options.

TheDodger committed:
Revert "Lets give this a try"

This reverts commit 002139e312b21404273a2440ddbb2e32f840d44f.

TheDodger committed:
Change resource limits and add corresponding information

TheDodger committed:
Wording

TheDodger committed:
Fix comment glitch

TheDodger committed:
Simply doesn't run at all if request too high

TheDodger committed:
Release
